xdsl/interpreter.py

```diff
@@ -30,16 +30,140 @@
 from xdsl.utils.exceptions import InterpretationError
 from xdsl.utils.scoped_dict import ScopedDict
 
-_IMPL_OP_TYPE = "__impl_op_type"
-_CAST_IMPL_TYPES = "__cast_impl_types"
-_ATTR_IMPL_TYPES = "__attr_impl_types"
-_EXT_FUNC_NAME = "__external_func_name"
-_CALLABLE_OP_TYPE = "__callable_op_type"
-_IMPL_DICT = "__impl_dict"
-_CAST_IMPL_DICT = "__cast_impl_dict"
-_ATTR_IMPL_DICT = "__attr_impl_dict"
-_EXT_FUNC_DICT = "__external_func_dict"
-_CALLABLE_IMPL_DICT = "__callable_impl_dict"
+PythonValues: TypeAlias = tuple[Any, ...]
+"""
+A tuple of result values corresponding to the results of the operation being
+interpreted.
+"""
```

Comment:
added some doc strings

Reply:
Otherwise just cut/paste from bottom of file, with the addition of `_IMPL_DICT` and friends, renamed the keys to have a KEY suffix.

Reply:
I have removed `_IMPL_DICT` and friends, now only change are copy/paste up, change names of keys to add `_KEY`, and add some doc strings.